frontend/src/components/Wallet.js

```diff
@@ -0,0 +1,184 @@
+			}) ;
+			setDetailData(details);
+			return;
+		}
+		const length = publicAddresses.length;
+		for (let i=0; i<length; i++) {
+			let pubAddress = publicAddresses[i];
+			const balance = await provider.request({ 
+				method: "eth_getBalance", 
+				params: [pubAddress, "latest"] 
+			});	// "Wei" balance in hex format, like 0x5a4e1804f198f1d99.
+			if (balance) {
+				details.push({ 
+					pubAddress: pubAddress,
+					balance: balance
+				});
+			}
+		}
+		setDetailData(details);
+	}
+
+	useEffect(() => {
+		fetchAccountsDetails(publicAddresses);
+	}, []);
+
+	return <TableContainer style={{ color: "white" }}>
+		<Table>
+			<TableBody>
+				<TableRow>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>
+						<Typography variant="subtitle1" gutterBottom component="div">
+							Account (Public Address)
+						</Typography>
+					</TableCell>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>
+						<Typography variant="subtitle1" gutterBottom component="div">
+							Balance (Eth)
+						</Typography>
+					</TableCell>
+				</TableRow>
+				{detailData.map((value, key) => {
+					return <AccountEntry key={key} data={value} />;
+				})}
+			</TableBody>
+		</Table>
+	</TableContainer>;
+}
+
+/**
+ * Component to display and add Eth address from Wallet. Currently only Metamask is supported.
+ * @param user user object of the currently logged in user/admin
+ * @returns {JSX.Element}
+ */
+export default function Wallet({ user }) {
+	const [loadedAddresses, setLoadedAddresses] = useState(false);
+	const [publicAddresses, setPublicAddresses] = useState([]);
+	const [errorNoMetamaskMessage, setErrorNoMetamaskMessage] = useState("");
+
+	const history = useHistory();
+
+	const routeChange = (path) =>{
+		history.push(path);
+	};
+
+	const getEthAddressesFromServer = () => {
+		if (!loadedAddresses) {
+			appwriteApi.getOwnEthAddress(user.$id)
+				.then(result => {
+					// Set the address(es)
+					let currentPubAddr = publicAddresses;
+					for (let i = 0; i < result.sum; i++) {
+						currentPubAddr.push(result.documents[i].walletAddress);
+					}
+					setPublicAddresses(currentPubAddr);
+					setLoadedAddresses(true);
+				})
+				.catch((error) => {
+					console.error(error);
+				});
+		}
+	};
+
+	useEffect(() => {
+		getEthAddressesFromServer();
+	}, []);
+
+	if (!user){
+		routeChange("/");
+		return <></>;
+	}
+
+	/* Handle button connect to MetaMask */
+	const handleAddMetaMask = async () => {
+		const ethereum = await detectEthereumProvider();
+		// Ref: https://docs.metamask.io/guide/rpc-api.html#table-of-contents
+		let pubAddr = 0;
+		if (ethereum) {
+			const accounts = await ethereum.request({ method: "eth_requestAccounts" });
+			const account = accounts[0];
+			pubAddr = account;
+			// Set address on server side
+			appwriteApi.setEthAddress(pubAddr);
+
+			let currentPubAddr = publicAddresses;
+			currentPubAddr.push(pubAddr);
+			setPublicAddresses(currentPubAddr);
```

Comment:
This does not have the expected behavior as you cannot edit a state this way. In fact, it does not work this way for me. Instead do this:
```suggestion
			setPublicAddresses(prevAddr => [...prevAddr, pubAddr]);
```

Reply:
Coincidentally, this is included in my code suggestion below :-) .

Reply:
Have you tested yours? You create an object, even though we need a list. I tested mine.

Reply:
I edited my example, I also saw the mistake :-) .

Reply:
fixed